mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoIncrementMethodGenerator.java

```diff
@@ -0,0 +1,191 @@
+    return ImmutableSet.of(VOID, FUTURE_OF_VOID, REACTIVE_RESULT_SET);
+  }
+
+  @Override
+  public boolean requiresReactive() {
+    // Validate the return type:
+    DaoReturnType returnType =
+        parseAndValidateReturnType(getSupportedReturnTypes(), Increment.class.getSimpleName());
+    if (returnType == null) {
+      return false;
+    }
+    return returnType.requiresReactive();
+  }
+
+  @Override
+  public Optional<MethodSpec> generate() {
+    // Validate the parameters:
+    // - the first one must be the entity.
+    // - a Function<BoundStatementBuilder, BoundStatementBuilder> can be added in second position.
+    List<? extends VariableElement> parameters = methodElement.getParameters();
+    VariableElement boundStatementFunction = findBoundStatementFunction(methodElement);
+    if (boundStatementFunction != null) {
+      parameters = parameters.subList(0, parameters.size() - 1);
+    }
+    TypeElement entityElement =
+        parameters.isEmpty()
+            ? null
+            : EntityUtils.asEntityElement(parameters.get(0), typeParameters);
+    if (entityElement == null) {
+      context
+          .getMessager()
+          .error(
+              methodElement,
+              processedType,
+              "%s methods must take the entity to update as the first parameter",
+              Increment.class.getSimpleName());
+      return Optional.empty();
+    }
+    warnIfCqlNamePresent(parameters.subList(0, 1));
+    EntityDefinition entityDefinition = context.getEntityFactory().getDefinition(entityElement);
+
+    // Validate the return type:
+    DaoReturnType returnType =
+        parseAndValidateReturnType(getSupportedReturnTypes(), Increment.class.getSimpleName());
+    if (returnType == null) {
+      return Optional.empty();
+    }
+
+    // Generate the method:
+    String helperFieldName = enclosingClass.addEntityHelperField(ClassName.get(entityElement));
+    String statementName =
+        enclosingClass.addPreparedStatement(
+            methodElement,
+            (methodBuilder, requestName) ->
+                generatePrepareRequest(
+                    methodBuilder, requestName, entityDefinition, helperFieldName));
+
+    CodeBlock.Builder updateStatementBlock = CodeBlock.builder();
+
+    updateStatementBlock.addStatement(
+        "$T boundStatementBuilder = $L.boundStatementBuilder()",
+        BoundStatementBuilder.class,
+        statementName);
+
+    populateBuilderWithStatementAttributes(updateStatementBlock, methodElement);
+    populateBuilderWithFunction(updateStatementBlock, boundStatementFunction);
+
+    String entityParameterName = parameters.get(0).getSimpleName().toString();
+
+    // All entity properties are bound in the generated request: primary key columns in the WHERE
+    // clause, and other columns in the SET clause.
+    updateStatementBlock.addStatement(
+        "$1L.set($2L, boundStatementBuilder, isProtocolVersionV3 ? $3T.$4L : $3T.$5L)",
```

Comment:
We do the same for regular `@Insert` and `@Update` methods, although it's less visible because it's buried in the entity helper's `set()` method.

I think it's more of a consequence of not having a special case for PK columns, but is it really a problem? If you try to pass an entity with null PK columns, the query will fail at runtime because those columns are absent, that's pretty consistent.